tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -28,7 +28,44 @@ use tracing_core::{
 
 /// A `Layer` which filters spans and events based on a set of filter
 /// directives.
-// TODO(eliza): document filter directive syntax?
+///
+/// # Directives
+///
+/// A filter consists of one or more directives. Directives match [`Span`]s and [`Event`]s
+/// and specify a maximum verbosity [`level`] to enable for those that match. The directive
+/// syntax is similar to the one presented in `env_logger`. The syntax consists
```

Comment:
I think we should link to `env_logger`'s docs on filtering at https://docs.rs/env_logger/0.7.1/env_logger/#enabling-logging